### PR TITLE
[docs] LoRA for token classification

### PR DESCRIPTION
**Pull request by stevhliu**
Moves the LoRA for token classification [notebook](https://github.com/huggingface/peft/blob/main/examples/token_classification/peft_lora_token_cls.ipynb) into the docs and adapts it for a less-complicated example instead of training LayoutLM on annotated forms.

- [ ] complete the training section with either a native PyTorch loop or `Trainer`

### CODE REVIEW

File: docs/source/task_guides/token-classification-lora.mdx [hunk 1]
```diff
@@ -0,0 +1,372 @@
...
+        "precision": results["overall_precision"],
+        "recall": results["overall_recall"],
+        "f1": results["overall_f1"],
+        "accuracy": results["overall_accuracy"],
+    }
+```
+
+## Preprocess dataset
+
+Initialize a tokenizer and make sure you set `is_split_into_words=True` because the text sequence has already been split into words. However, this doesn't mean it is tokenized yet (even though it may look like it!), and you'll need to further tokenize the words into subwords.
```
Comment (pacman100) [inline docs/source/task_guides/token-classification-lora.mdx:153]: This was something new for me, thanks for highlighting this!

File: docs/source/task_guides/token-classification-lora.mdx [hunk 2]
```diff
@@ -0,0 +1,372 @@
...
+
+- `task_type`, token classification
+- `r`, the dimension of the low-rank matrices
+- `lora_alpha`, scaling factor for the weight matrices
+- `lora_dropout`, dropout probability of the LoRA layers
+- `bias`, set to `all` to train all bias parameters
+
+<Tip>
+
+💡 The weight matrix is scaled by `lora_alpha/r` so changing one or the other will affect the number of trainable parameters. In the paper, the authors set `lora_alpha` equal to `r` to reduce the need to tune the hyperparameters when changing the value of `r`.
```
Comment (pacman100) [inline docs/source/task_guides/token-classification-lora.mdx:243]: Actually, the delta activations from LoRA are scaled as per alpha/r. So, higher alpha means that higher weightage is given to the LoRA activations and as such the gradients are also scaled accordingly. MAM paper uses LoRA and Scaled Parallel Adapter with alpha being 4X compared to r. The LoRA paper itself uses alpha ranging from 1X-8X that of r. 

Alpha doesn't change the number of trainable parameters, rather functions as a scaling factor.

Comment (pacman100) [reply docs/source/task_guides/token-classification-lora.mdx:243]: Recommended way would be to first see the performance without setting bias, so that pretrained model is completely frozen to keep the generalization of it intact. If te performance doesn't meet the requirements, then they can experiment with setting bias to `lora_only` and as a final step to `all`

File: docs/source/task_guides/token-classification-lora.mdx [hunk 3]
```diff
@@ -0,0 +1,372 @@
...
+
+Finally, create a data collator to pad the examples to longest length in a batch:
+
+```py
+data_collator = DataCollatorForTokenClassification(tokenizer=tokenizer)
+```
+
+## Train
+
+Now you can create a [`PeftModel`]. Start by loading the base `bloom-560m` model, the number of expected labels, and the `id2label` and `label2id` dictionaries:
```
Comment (pacman100) [inline docs/source/task_guides/token-classification-lora.mdx:198]:
```suggestion
Now you can create a [`PeftModel`]. Start by loading the base `roberta-large` model, the number of expected labels, and the `id2label` and `label2id` dictionaries:
```

**Review by pacman100 — COMMENTED** [review]
Thanks a lot @stevhliu for simplifying the token classification example to an interesting bionlp dataset! 

Left few comments

**Review by MKhalusova — APPROVED** [review]
Nice guide! I left a few nits here and there.

File: docs/source/task_guides/token-classification-lora.mdx (hunk 2 above)
Comment (stevhliu) [reply docs/source/task_guides/token-classification-lora.mdx:243]: Really interesting, thanks for this!

File: docs/source/task_guides/token-classification-lora.mdx [hunk 4]
```diff
@@ -0,0 +1,372 @@
...
+    task_type=TaskType.TOKEN_CLS, inference_mode=False, r=16, lora_alpha=16, lora_dropout=0.1, bias="all"
+)
+```
+
+Then pass the base model and `peft_config` to the [`get_peft_model`] function to create a [`PeftModel`]. You can check out how much more efficient the [`PeftModel`] is compared to fully training the base model by printing out the trainable parameters:
+
+```py
+model = get_peft_model(model, peft_config)
+model.print_trainable_parameters()
+"trainable params: 1856523 || all params: 560798731 || trainable%: 0.3310497862021731"
```
[the remark this answers is not in the document]
Comment (stevhliu) [reply docs/source/task_guides/token-classification-lora.mdx:258]: Yes, forgot to update it! :)

**Review by pacman100 — APPROVED** [review]
Thank you @stevhliu for iterating, LGTM! 🚀
I really like the example and the dataset used.
